README.md

```diff
@@ -105,7 +145,38 @@ httpClient.execute(request: request, deadline: .now() + .milliseconds(1))
 ```
 
 ### Streaming
-When dealing with larger amount of data, it's critical to stream the response body instead of aggregating in-memory. Handling a response stream is done using a delegate protocol. The following example demonstrates how to count the number of bytes in a streaming response body:
+When dealing with larger amount of data, it's critical to stream the response body instead of aggregating in-memory. 
+The following example demonstrates how to count the number of bytes in a streaming response body:
+
+#### Using Swift Concurrency
+```swift
+let httpClient = HTTPClient(eventLoopGroupProvider: .createNew)
+
+let request = HTTPClientRequest(url: "https://apple.com/")
+let response = try await httpClient.execute(request, timeout: .seconds(30))
+print("HTTP head", response)
+
+// if defined, the content-length headers announces the size of the body
+let expectedBytes = response.headers.first(name: "content-length").flatMap(Int.init)
+
+var receivedBytes = 0
+// asynchronously iterates over all body fragments.
+for try await buffer in response.body {
+    // For this example, we are just interested in the size of the fragment
+    receivedBytes += buffer.readableBytes
+    
+    if let expectedBytes = expectedBytes {
+        // if the body size is known, we calculate a progress indicator
+        let progress = Double(receivedBytes)/Double(expectedBytes)
+        print("progress: \(Int(progress * 100))%")
+    }
+    // in case backpressure is needed, all reads will be paused until the end of the for loop.
```

Comment:
backpressure has nothing todo with the loop, and I don't think we should explain it here. It just works ;)

Reply:
I think we should mentioned this at least somewhere. AFAIK, this is not documented anywhere or am I missing something? This is a really great feature and we should definitely mention it.

Reply:
I'm with @dnadoba that we should mention it, but I'm with @fabianfett that I don't think we need to explain it: it just works. In this case, maybe at the top of the loop I'd change that comment to something like:

```
// asynchronously iterates over all body fragments
// this loop will automatically propagate backpressure correctly
```